### PR TITLE
Operation Thin crust : Nanotransen Pizza delivery

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

You can order Pizza from Centcom for 20k credits.
You order to a location and the Pizza Delivery Guy arrives and gives you your pizza.

## Why It's Good For The Game

We need more pizza

## Changelog
:cl:
add: The Crew can now order Pizza for 20'000 credits from a comms console
add: Pizza Delivery ERT. Send these boys in when the deathsquad dies.
fix: Added access requirement to a Centcom door so Inspectors can't run around inside centcom proper
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
give 'em a giant rock to drive arounf

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -329,6 +329,20 @@
 				usr.log_message("has requested the nuclear codes from CentCom with reason \"[input]\"", LOG_SAY)
 				priority_announce("The codes for the on-station nuclear self-destruct have been requested by [usr]. Confirmation or denial of this request will be sent shortly.", "Nuclear Self Destruct Codes Requested",'sound/ai/commandreport.ogg')
 				CM.lastTimeUsed = world.time
+		if("pizza")	//OPERATION BREADSTICK IS A GO
+			if(authenticated)
+				var/input = stripped_input(usr, "Please enter the location for delivering the Pizza. Delivery not guaranteed.", "Pizza Request.","")
```
Comment (trollbreeder) [inline code/game/machinery/computer/communications.dm:334]: why is pizza capitalized

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -329,6 +329,20 @@
 				usr.log_message("has requested the nuclear codes from CentCom with reason \"[input]\"", LOG_SAY)
 				priority_announce("The codes for the on-station nuclear self-destruct have been requested by [usr]. Confirmation or denial of this request will be sent shortly.", "Nuclear Self Destruct Codes Requested",'sound/ai/commandreport.ogg')
 				CM.lastTimeUsed = world.time
+		if("pizza")	//OPERATION BREADSTICK IS A GO
+			if(authenticated)
+				var/input = stripped_input(usr, "Please enter the location for delivering the Pizza. Delivery not guaranteed.", "Pizza Request.","")
+				if(!input || !(usr in view(1,src)))
+					return
+				var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+				if(D.account_balance <= 20000)	//so people dont spam pizzamen
```
Comment (ATH1909) [inline code/game/machinery/computer/communications.dm:338]: should just be <, not <=

File: code/game/machinery/computer/communications.dm [hunk 3]
```diff
@@ -329,6 +329,20 @@
...
 				priority_announce("The codes for the on-station nuclear self-destruct have been requested by [usr]. Confirmation or denial of this request will be sent shortly.", "Nuclear Self Destruct Codes Requested",'sound/ai/commandreport.ogg')
 				CM.lastTimeUsed = world.time
+		if("pizza")	//OPERATION BREADSTICK IS A GO
+			if(authenticated)
+				var/input = stripped_input(usr, "Please enter the location for delivering the Pizza. Delivery not guaranteed.", "Pizza Request.","")
+				if(!input || !(usr in view(1,src)))
+					return
+				var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+				if(D.account_balance <= 20000)	//so people dont spam pizzamen
+					to_chat(usr, "<span class='warning'>You don't have enough money to make a Pizza delivery!</span>")
```
Comment (ATH1909) [inline code/game/machinery/computer/communications.dm:339]: *order

**Comment by ATH1909** [comment]
time to order a pizza, mug the delivery guy for his ID and shuttle, then raid centcom for ERT gear

I approve, actually, especially considering that this costs 20,000 credits. It's practically a station project.

File: code/game/machinery/computer/communications.dm (hunk 1 above)
Comment (Jack7D1) [reply code/game/machinery/computer/communications.dm:334]: All hail the mighty Pizza

Comment (TiviPlus) [reply code/game/machinery/computer/communications.dm:334]: How dare thee insult The Mighty Pizza

**Comment by Shadowflame909** [comment]
If they don't give you your pizza in 15 minutes it should be free

**Comment by LordSaladin** [comment]
thermite solves everything.

**Comment by TiviPlus** [comment]
Have fun thermiting through adminwalls

**Comment by optimumtact** [comment]
I don't know if I'll merge this, since centcomm is traditional admin only

**Comment by Arianya** [comment]
Currently testmerged on Sybil

**Comment by Arianya** [comment]
I will say that, yes, this shouldn't be a 20k "buy a pass to centcomm" thing - either the Pizza Guy's ID should only lead to a small pizza delivery section. with no doors out to greater centcomm, or the pizza guy should be a one way trip - what does happen to them anyway after they deliver?

**Comment by TiviPlus** [comment]
> I will say that, yes, this shouldn't be a 20k "buy a pass to centcomm" thing - either the Pizza Guy's ID should only lead to a small pizza delivery section. with no doors out to greater centcomm, or the pizza guy should be a one way trip - what does happen to them anyway after they deliver?

Agreed, thats why I added an access requirement to the Centcom shutters so that they cant go into Centcom proper. When they deliver they're free to go and RP with the crew, and they have a pretty sizable supply of pizza so they can find someone else to deliver to.

**Comment by optimumtact** [comment]
I think it would be best if it's a one way trip

**Comment by TiviPlus** [comment]
Agreed, I'll do it once I finish other stuff and fix my tg branches

File: code/modules/clothing/outfits/ert.dm [hunk 4]
```diff
@@ -420,3 +420,32 @@
...
+/datum/outfit/centcom/ert/pizza/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	if(visualsOnly)
+		return
+
+	var/obj/item/card/id/W = H.wear_id
+	W.access = get_centcom_access("CentCom Official")
+	W.access += ACCESS_WEAPONS
+	W.assignment = "Pizza Deivery Guy"
+	W.update_label()
+	..()
```
Comment (Rohesie) [inline code/modules/clothing/outfits/ert.dm:451]:
```suggestion
	return ..()
```

File: code/game/machinery/computer/communications.dm [hunk 5]
```diff
@@ -329,6 +329,20 @@
 				usr.log_message("has requested the nuclear codes from CentCom with reason \"[input]\"", LOG_SAY)
 				priority_announce("The codes for the on-station nuclear self-destruct have been requested by [usr]. Confirmation or denial of this request will be sent shortly.", "Nuclear Self Destruct Codes Requested",'sound/ai/commandreport.ogg')
 				CM.lastTimeUsed = world.time
+		if("pizza")	//OPERATION BREADSTICK IS A GO
+			if(authenticated)
+				var/input = stripped_input(usr, "Please enter the location for delivering the pizza. Delivery not guaranteed.", "Pizza Request.","")
+				if(!input || !(usr in view(1,src)))
```
Comment (Rohesie) [inline code/game/machinery/computer/communications.dm:335]:
```suggestion
				if(!input || !user.Adjacent(src))

```

**Comment by TiviPlus** [comment]
I have IRL stuff to do so I cant redo this PR atm, maybe I'll finish it later if I'm still interested

**Pull request closed without merging**